ngraph/frontend/paddlepaddle/src/op/lstm.cpp

```diff
@@ -0,0 +1,212 @@
+                                                                        batch_size_node);
+
+                            auto init_states = node.get_ng_inputs("PreState");
+                            // 0 for init_h, 1 for init_cell, update bidirect_len for init states
+                            bidirect_len = node.get_attribute<bool>("is_bidirec") ? 2 : 1;
+
+                            auto h_begin = opset6::Constant::create(element::Type_t::i64, {1}, {layer * bidirect_len});
+                            auto h_end = opset6::Constant::create(element::Type_t::i32, Shape{1}, {layer * bidirect_len + bidirect_len});
+                            auto c_begin = opset6::Constant::create(element::Type_t::i64, {1}, {layer * bidirect_len});
+                            auto c_end = opset6::Constant::create(element::Type_t::i64, {1}, {layer * bidirect_len + bidirect_len});
+
+                            m_input_map[LSTMInput::LSTM_INPUT_INIT_H] = builder::opset1::reorder_axes(std::make_shared<opset6::StridedSlice>(init_states[0], h_begin, h_end, std::vector<int64_t>{0}, std::vector<int64_t>{0}), {1, 0, 2});
+                            m_input_map[LSTMInput::LSTM_INPUT_INIT_C] = builder::opset1::reorder_axes(std::make_shared<opset6::StridedSlice>(init_states[1], c_begin, c_end,std::vector<int64_t>{0}, std::vector<int64_t>{0}), {1, 0, 2});
+
+                        }
+
+                        Output<ngraph::Node> &at(const LSTMInput &key) { return m_input_map.at(key); }
+
+                        std::map<LSTMInput, Output<ngraph::Node>> m_input_map;
+                    };
+
+                    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ATTRIBUTES PARSING ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                    struct LSTMAttributes
+                    {
+                        explicit LSTMAttributes(const NodeContext &node)
+                                : m_hidden_size(node.get_attribute<int32_t>("hidden_size")),
+                                  m_layers(node.get_attribute<int32_t>("num_layers")),
+                                  m_direction(node.get_attribute<bool>("is_bidirec") ? ngraph::op::RecurrentSequenceDirection::BIDIRECTIONAL : ngraph::op::RecurrentSequenceDirection::FORWARD)
+                        {};
+
+                        ngraph::op::RecurrentSequenceDirection m_direction;
+                        std::int32_t m_hidden_size;
+                        std::int32_t m_layers;
+
+                    };
+                }
+                NamedOutputs lstm (const NodeContext& node) {
+                    auto mode = node.get_attribute<std::string>("mode");
+                    PDPD_ASSERT(mode == "LSTM", "RNN only support LSTM now");
+                    auto prev_inputs = node.get_ng_inputs("Input");
+                    Output<Node> prev_output = prev_inputs[0];
+                    LSTMAttributes attrs(node);
+                    Output<Node> Y_h, Y_c;
+                    for(int i = 0; i < attrs.m_layers; i++)
```

Comment:
According to PDPD spec (https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/layer/rnn/LSTM_en.html#lstm) Lstm is an equivalent to OpenVINO LSTM Sequence. LSTMSequence is already sequence of LSTM units (LSTMCells), `for `loop is unnecessary

Reply:
Hi I check the source code and find OV LSTMSequence can only handle part of situations. The PDPD LSTM has a *layers* parameter, when PDPD LSTM has bidirection LSTM with layers more than 1, OV version cannot handle this situation. Reference:
https://github.com/PaddlePaddle/Paddle/blob/bd2a4e23fad95fbd10f5ea33f1ab922b65be6515/python/paddle/nn/layer/rnn.py#L905

Reply:
### OV LSTMSequence:
1: X - 3D tensor of type T1 [batch_size, seq_length, input_size], input data. It differs from LSTMCell 1st input only by additional axis with size seq_length. Required.
2: initial_hidden_state - 3D tensor of type T1 **[batch_size, num_directions, hidden_size]**, input hidden state data. Required.
3: initial_cell_state - 3D tensor of type T1 **[batch_size, num_directions, hidden_size]**, input cell state data. Required.
### PDPD LSTM:
1. inputs (Tensor): the input sequence. If time_major is True, the shape is [time_steps, batch_size, input_size], else, the shape is [batch_size, time_steps, hidden_size].

2. initial_states (tuple, optional): the initial state, a tuple of (h, c), the shape of each is **[num_layers * num_directions, batch_size, hidden_size]**. If initial_state is not given, zero initial states are used.

I think the **layers** makes LSTM different from OV LSTMSequence

Reply:
Thanks for the detailed explanation!
I would recommend to use OV TensorIterator layer to wrap LSTMSequence. This layer already has functionality to slice inputs and iterate along the slices, and concatenate the results. It also allow to describe the connections between iterations. https://docs.openvinotoolkit.org/latest/openvino_docs_ops_infrastructure_TensorIterator_1.html

Reply:
Got it, will leave a TODO here as a reminder